crates/astria-sequencer-relayer/src/relayer/write.rs

```diff
@@ -189,18 +202,24 @@ impl BlobSubmitter {
     pub(super) async fn run(mut self) -> eyre::Result<()> {
         let mut submission = Fuse::terminated();
 
+        // Fetch the latest Celestia height so there is a starting point when tracking
+        // submissions.
+        fetch_latest_celestia_height(self.client.clone(), self.state.clone())
+            .await
+            .wrap_err("failed fetching latest celestia height after many retries")?;
+
         loop {
             select!(
                 Some(block) = self.blocks.recv(), if self.has_capacity() => {
```

Comment:
just double checking - `if self.has_capacity()` is called/checked before `self.blocks.recv()` right? so if there is no capacity, `block` will not be taken out of the channel and dropped, but only taken out when there is capacity?

Reply:
Yeah, the `select!` macro first evaluates the guard, if present. It then polls the futures to the left of the guard if and only if the guard evaluates to `true`.

So in this case the channel will only be polled if the submitter thinks it has capacity.

Have a read here: https://docs.rs/tokio/latest/tokio/macro.select.html